packages/sandpack-core/src/npm/merge-dependency.ts

```diff
@@ -163,20 +165,27 @@ export function mergeDependencies(responses: ILambdaResponse[]) {
         d => d.name === depDepName
       );
 
-      if (
-        rootDependency &&
-        !intersects(rootDependency.version, newDepDep.semver) &&
-        // If the resolved or semver is the same we don't have to actually replace dependency info
-        rootDependency.version !== newDepDep.resolved &&
-        rootDependency.version !== newDepDep.semver &&
-        rootDependency.name !== r.dependency.name // and this dependency doesn't require an older version of itself
-      ) {
-        // If a root dependency is in conflict with a child dependency, we always
-        // go for the root dependency
-        replaceDependencyInfo(r, depDepName, newDepDep);
-
-        // Start from the beginning, to make sure everything is correct
-        i = -1;
+      if (rootDependency) {
+        // packages that require themselves?
+        if (r.dependency.name !== depDepName) {
+          if (!intersects(rootDependency.version, newDepDep.semver)) {
+            // If a root dependency is in conflict with a child dependency, we always
+            // go for the root dependency
+            replaceDependencyInfo(r, depDepName, newDepDep);
+
+            // Start from the beginning, to make sure everything is correct
+            i = -1;
+          } else {
+            // Remove the contents so we don't overwrite the root version's content
+            const pathPrefix = `/node_modules/${depDepName}`;
+            Object.keys(r.contents).forEach(p => {
+              if (p.startsWith(pathPrefix) && !VERSIONED_MODULE_RE.test(p)) {
+                delete r.contents[p];
```

Comment:
Not entirely sure if this is safe to do, do both responses return all exports or is the contents specific to the dependency?

Reply:
Ahh, interesting. So the root problem was that we were overwriting content which shouldn't have been overwritten?

Reply:
Yup, that was the problem